### PR TITLE
feat: add unstake screen for mobile staking

### PR DESCRIPTION
**Pull request by amitabh94**
## **Description**

This PR covers the following:

* Adds new unstake screen 
* Uses the existing components with the same Max, %age, error states and currency conversion toggle
* Adds unstaking information banner
* Adds a new hook file for input screens 

## **Related issues**

Closes [STAKE- 783](https://consensyssoftware.atlassian.net/browse/STAKE-783)

## **Manual testing steps**

1. Add export MM_POOLED_STAKING_UI_ENABLED=true to your .js.env file.
2. Go to  the ETH Token details Page
3. Click on the Unstake Button 
4. You should be navigated to unstake screen.

### **Before**

https://github.com/user-attachments/assets/51715c21-26b8-4eaa-bc87-a725282f85b2

### **After**

https://github.com/user-attachments/assets/428c4e17-4b15-447a-8c6c-83e8704be91a

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Stake/hooks/useStakingInput.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+  const handleCurrencySwitch = useCallback(() => {
+    setIsEth(!isEth);
+  }, [isEth]);
+
+  const percentageOptions = [
+    { value: 0.25, label: '25%' },
+    { value: 0.5, label: '50%' },
+    { value: 0.75, label: '75%' },
+    { value: 1, label: 'Max' },
```
Comment (Matt561) [inline app/components/UI/Stake/hooks/useStakingInput.ts:85]: Should we break `Max` out into `en.json` for translation support?
```suggestion
    { value: 1, label: strings('stake.max') },
```

File: app/components/UI/Stake/hooks/useStakingInput.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  ];
+
+  const handleAmountPress = useCallback(
+    ({ value }: { value: number }) => {
+      if (!balance) return;
+      const percentage = value * 100;
+      const amountPercentage = balance
+        ?.mul(new BN(percentage))
```
Comment (Matt561) [inline app/components/UI/Stake/hooks/useStakingInput.ts:93]: Is this optional chain needed if we return early when `!balance`?

```suggestion
        .mul(new BN(percentage))
```

**Review by Matt561 — COMMENTED** [review]
Left some suggestions but otherwise LGTM! 💯

File: app/components/UI/Navbar/index.js [hunk 3]
```diff
@@ -1825,7 +1824,7 @@ export const getSettingsNavigationOptions = (title, themeColors) => {
   };
 };
 
-export function getStakeInputNavbar(navigation, themeColors) {
+export function getStakingNavbar(title, navigation, themeColors) {
```
[the remark this answers is not in the document]
Comment (amitabh94) [reply app/components/UI/Navbar/index.js:1827]: Let's keep it consistent in all screens in line with other experiences as well. Will check in with design team to see if there is a strong reason to have different headers for different screens.

File: app/components/UI/Stake/hooks/useStakingInput.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  renderFromTokenMinimalUnit,
+  fiatNumberToWei,
+  fromTokenMinimalUnitString,
+  limitToMaximumDecimalPlaces,
+  renderFiat,
+} from '../../../../util/number';
+
+const useStakingInputHandlers = (balance: BN) => {
+  const [amount, setAmount] = useState('0');
+  const [amountBN, setAmountBN] = useState<BN>(new BN(0));
```
[the remark this answers is not in the document]
Comment (amitabh94) [reply app/components/UI/Stake/hooks/useStakingInput.ts]: Yeah that makes sense!

File: app/components/UI/Stake/hooks/useStakingInput.ts (hunk 2 above)
Comment (amitabh94) [reply app/components/UI/Stake/hooks/useStakingInput.ts]: Agreed! Not required.

**Review by Matt561 — APPROVED** [review]

**Review by EtherWizard33 — APPROVED** [review]
LGTM

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**